types/query/pagination.go

```diff
@@ -3,13 +3,40 @@ package query
 import (
 	"fmt"
 
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
 	"github.com/cosmos/cosmos-sdk/store/types"
 )
 
 // DefaultLimit is the default `limit` for queries
 // if the `limit` is not supplied, paginate will use `DefaultLimit`
 const DefaultLimit = 100
 
+// ParsePagination validate PageRequest and returns page number & limit.
+func ParsePagination(pageReq *PageRequest) (page, limit int, err error) {
```

Comment:
Nice. Can `x/auth/tx`'s GetTxsEvent use this function? or are the 2 pagination logics a bit different?

Reply:
Yes, `GetTxsEvent` has the same logic.

Reply:
Cool. Would you like to change GetTxsEvent to use this function in this PR, or in #8029 ?

Reply:
I'll address it in #8029.